data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarderProvider.java

```diff
@@ -35,25 +36,26 @@ public PeerForwarder register(final String pipelineName, final String pluginId,
             throw new RuntimeException("Data Prepper 2.0 will only support a single peer-forwarder per pipeline/plugin type");
         }
 
-        final PeerForwarderReceiveBuffer<Record<?>> peerForwarderReceiveBuffer = createBufferPerPipelineProcessor(pipelineName, pluginId);
-        // TODO: pass buffer to RemotePeerForwarder constructor
+        final PeerForwarderReceiveBuffer<Record<Event>> peerForwarderReceiveBuffer = createBufferPerPipelineProcessor(pipelineName, pluginId);
 
         if (isPeerForwardingRequired()) {
             if (hashRing == null) {
                 hashRing = peerForwarderClientFactory.createHashRing();
             }
-            return new RemotePeerForwarder(peerForwarderClient, hashRing, pipelineName, pluginId, identificationKeys);
+            return new RemotePeerForwarder(
+                    peerForwarderClient, hashRing, peerForwarderReceiveBuffer, pipelineName, pluginId, identificationKeys
+            );
         }
         else {
             return new LocalPeerForwarder();
         }
     }
 
-    private PeerForwarderReceiveBuffer<Record<?>> createBufferPerPipelineProcessor(final String pipelineName, final String pluginId) {
-        final PeerForwarderReceiveBuffer<Record<?>> peerForwarderReceiveBuffer = new
+    private PeerForwarderReceiveBuffer<Record<Event>> createBufferPerPipelineProcessor(final String pipelineName, final String pluginId) {
```

Comment:
Is the change from wildcard to `Event` because we only support `Event` type and no longer support `ExportTraceServiceRequest`?

Reply:
Core peer forwarder only supports internal events, as forwarded events won't go through the source again. So when forwarder they're still events. 
In peer forwarder plugin we used forwarded `ExportTraceServiceRequest` and each request has to be converted to event at source again.